sklearn/model_selection/_split.py

```diff
@@ -913,6 +920,165 @@ def get_n_splits(self, X, y, groups):
         return int(comb(len(np.unique(groups)), self.n_groups, exact=True))
 
 
+class _RepeatedSplits(with_metaclass(ABCMeta)):
+    """Repeated splits for an arbitrary randomized CV splitter.
+
+    Repeats splits for cross-validators n times with different randomization
+    in each repetition.
+
+    Parameters
+    ----------
+    cv : callable
+        Constructor of cross-validator.
+
+    n_repeats : int, default=5
+        Number of times cross-validator needs to be repeated.
+
+    random_state : None, int or RandomState, default=None
+        Random state to be used to generate random state for each
+        repetition.
+
+    **cvargs : additional params
+        Constructor parameters for cv. Must not contain random_state
+        and shuffle.
+    """
+    def __init__(self, cv, n_repeats=5, random_state=None, **cvargs):
+        if not isinstance(n_repeats, (np.integer, numbers.Integral)):
+            raise ValueError("Number of repetitions must be of Integral type.")
+
+        if n_repeats <= 1:
+            raise ValueError("Number of repetitions must be greater than 1.")
+
+        self.cv = cv
+        self.n_repeats = n_repeats
+        self.random_state = random_state
+        self.cvargs = cvargs
+
+    def split(self, X, y=None, groups=None):
+        """Generates indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        y : array-like, of length n_samples
+            The target variable for supervised learning problems.
+
+        groups : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+        Returns
+        -------
+        train : ndarray
+            The training set indices for that split.
+
+        test : ndarray
+            The testing set indices for that split.
+        """
+        n_repeats = self.n_repeats
+        rng = check_random_state(self.random_state)
+
+        for idx in range(n_repeats):
+            random_state = rng.randint(np.iinfo(np.int32).max)
```

Comment:
Your code is similar to that one  :
```python
In [1]: from sklearn.utils import check_random_state

In [2]: class Foo:
   ...:     def __init__(self, random_state):
   ...:         rng = check_random_state(random_state)
   ...:         
   ...:     def split(self):
   ...:         print(rng.randint(1000))
   ...:         

In [3]: rng = check_random_state(0)

In [4]: f1 = Foo(rng)

In [5]: f2 = Foo(rng)

In [6]: f1.split()
684

In [7]: f1.split()
559

In [8]: f2.split()
629

In [9]: f2.split()
192
```
As you can see values are different at each iteration and for different instance.
`rng` is an object and it will be modified by self.cv at each iteration.